src/phase_change/EnthalpyHierarchyIntegrator.cpp

```diff
@@ -0,0 +1,1094 @@
+#include "ibamr/AdvDiffConvectiveOperatorManager.h"
+#include "ibamr/AdvDiffHierarchyIntegrator.h"
+#include "ibamr/AdvDiffSemiImplicitHierarchyIntegrator.h"
+#include "ibamr/EnthalpyHierarchyIntegrator.h"
+#include "ibamr/ibamr_enums.h"
+#include "ibamr/ibamr_utilities.h"
+
+#include "ibtk/CCLaplaceOperator.h"
+#include "ibtk/CartGridFunction.h"
+#include "ibtk/IBTK_MPI.h"
+#include "ibtk/LaplaceOperator.h"
+#include "ibtk/PoissonSolver.h"
+
+#include "BasePatchHierarchy.h"
+#include "CartesianGridGeometry.h"
+#include "CartesianPatchGeometry.h"
+#include "CellDataFactory.h"
+#include "CellVariable.h"
+#include "FaceData.h"
+#include "FaceVariable.h"
+#include "GriddingAlgorithm.h"
+#include "HierarchyCellDataOpsReal.h"
+#include "HierarchyDataOpsManager.h"
+#include "HierarchyFaceDataOpsReal.h"
+#include "HierarchySideDataOpsReal.h"
+#include "IntVector.h"
+#include "Patch.h"
+#include "PatchFaceDataOpsReal.h"
+#include "PatchHierarchy.h"
+#include "PatchLevel.h"
+#include "PoissonSpecifications.h"
+#include "SideVariable.h"
+#include "Variable.h"
+#include "VariableContext.h"
+#include "VariableDatabase.h"
+#include "tbox/Database.h"
+#include "tbox/MathUtilities.h"
+#include "tbox/MemoryDatabase.h"
+#include "tbox/PIO.h"
+#include "tbox/Pointer.h"
+#include "tbox/RestartManager.h"
+#include "tbox/Utilities.h"
+
+#include <string>
+#include <vector>
+
+#include "ibamr/namespaces.h" // IWYU pragma: keep
+
+namespace SAMRAI
+{
+namespace hier
+{
+template <int DIM>
+class Box;
+} // namespace hier
+} // namespace SAMRAI
+
+namespace SAMRAI
+{
+namespace solv
+{
+template <int DIM>
+class RobinBcCoefStrategy;
+} // namespace solv
+} // namespace SAMRAI
+
+/////////////////////////////// NAMESPACE ////////////////////////////////////
+
+namespace IBAMR
+{
+/////////////////////////////// STATIC ///////////////////////////////////////
+namespace
+{
+// Version of INSHierarchyIntegrator restart file data.
+static const int ENTHALPY_PC_HIERARCHY_INTEGRATOR_VERSION = 4;
+
+// Number of ghosts cells used for each variable quantity.
+static const int CELLG = 1;
+static const int NOGHOSTS = 0;
+
+static const double H_LIM = 0.5;
+} // namespace
+
+/////////////////////////////// PUBLIC ///////////////////////////////////////
+
+EnthalpyHierarchyIntegrator::EnthalpyHierarchyIntegrator(const std::string& object_name,
+                                                         Pointer<Database> input_db,
+                                                         bool register_for_restart)
+    : PhaseChangeHierarchyIntegrator(object_name, input_db, register_for_restart)
+{
+#if !defined(NDEBUG)
+    TBOX_ASSERT(!object_name.empty());
+    TBOX_ASSERT(input_db);
+#endif
+
+    // Initialize object with data read from the input and restart databases.
+    bool from_restart = RestartManager::getManager()->isFromRestart();
+
+    if (from_restart) getFromRestart();
+    getFromInput(input_db, from_restart);
```

Comment:
This would be clearer if you flipped these around and encoded the default values in `getFromInput`. The same goes for the other classes - the input database always happens first, then restart data.